### PR TITLE
Updated cart

### PR DESCRIPTION
**Pull request by DenisZaviryuhin**

### CODE REVIEW

File: src/main/java/com/example/hop_oasis/controller/CartController.java [hunk 1]
```diff
@@ -1,93 +1,51 @@
...
 import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.List;
 
 @RestController
-@RequiredArgsConstructor
 @RequestMapping("/cart")
```
Comment (vmakar1134) [inline src/main/java/com/example/hop_oasis/controller/CartController.java:15]: /carts

File: src/main/java/com/example/hop_oasis/controller/CartController.java [hunk 2]
```diff
@@ -1,93 +1,51 @@
...
+    public ResponseEntity<CartItemDto> addItem(@RequestBody ItemRequestDto itemRequestDto) {
+        CartItemDto dto = cartService.add(itemRequestDto.getItemId(), itemRequestDto.getQuantity(), itemRequestDto.getItemType());
         return ResponseEntity.ok(dto);
     }
 
-    @PutMapping
-    public ResponseEntity<CartItemDto> updateQuantity(@RequestBody ItemRequestDto itemRequest) {
-        CartItemDto dto = cartService.updateQuantity(itemRequest.getItemId(), itemRequest.getQuantity(), itemRequest.getItemType());
-        return ResponseEntity.ok(dto);
+    @PutMapping("/update")
```
Comment (vmakar1134) [inline src/main/java/com/example/hop_oasis/controller/CartController.java:33]: `/update` is redundant. PUT method by itself means that the resource is updated

File: src/main/java/com/example/hop_oasis/controller/CartController.java [hunk 3]
```diff
@@ -1,93 +1,51 @@
...
-//    public ResponseEntity<CartItemDto> add(@RequestBody ItemRequestDto itemRequest) {
-//        CartItemDto dto = cartService.add(itemRequest.getItemId(), itemRequest.getQuantity(), itemRequest.getItemType());
-//        return ResponseEntity.ok(dto);
-//    }
-    @GetMapping("/items")public ResponseEntity<CartItemDto> addItem(
-            @RequestParam("itemId") Long itemId,
-            @RequestParam("quantity") int quantity,
-            @RequestParam("itemType") ItemType itemType) {
-        CartItemDto dto = cartService.add(itemId, quantity, itemType);
+    @PostMapping("/add")
```
Comment (vmakar1134) [inline src/main/java/com/example/hop_oasis/controller/CartController.java:27]: `/add` is redundant. POST method by itself means that the resource is created

File: src/main/java/com/example/hop_oasis/controller/CartController.java [hunk 4]
```diff
@@ -1,93 +1,51 @@
...
 import org.springframework.web.bind.annotation.*;
 
-import java.util.List;
 
 @RestController
-@RequiredArgsConstructor
 @RequestMapping("/cart")
-@SessionAttributes("cart")
+@RequiredArgsConstructor
+@Log4j2
```
Comment (vmakar1134) [inline src/main/java/com/example/hop_oasis/controller/CartController.java:17]: You can remove @Log4j2 if you don't log anything

File: src/main/java/com/example/hop_oasis/controller/CartController.java (hunk 1 above)
Comment (DenisZaviryuhin) [reply src/main/java/com/example/hop_oasis/controller/CartController.java:15]: done

File: src/main/java/com/example/hop_oasis/controller/CartController.java (hunk 2 above)
Comment (DenisZaviryuhin) [reply src/main/java/com/example/hop_oasis/controller/CartController.java:33]: done

File: src/main/java/com/example/hop_oasis/controller/CartController.java (hunk 3 above)
Comment (DenisZaviryuhin) [reply src/main/java/com/example/hop_oasis/controller/CartController.java:27]: done

File: src/main/java/com/example/hop_oasis/controller/CartController.java (hunk 4 above)
Comment (DenisZaviryuhin) [reply src/main/java/com/example/hop_oasis/controller/CartController.java:17]: done

**Review by vmakar1134 — APPROVED** [review]

**Pull request merged**